### ISSUE TITLE
Highcharts boost draws empty chart after redraw

### ISSUE THREAD

**Issue by soetji** [highcharts/highcharts#12554]
Highcharts draws no lines when boost is enabled. No problem when boost is disabled. Looks like it has to do with calling yAxis.setTitle() in the chart.render event. Boosts draws properly if yAxis.setTitle() is not called.

I need to set the y-axis title dynamically based on the chart's dynamic data.

#### Expected behaviour
Draw lines

#### Actual behaviour
No lines

#### Live demo with steps to reproduce
Only for testing I set the boost seriesThreshold to 1.
https://jsfiddle.net/soetji/m5vLgk0q/

#### Product version
Latest on https://code.highcharts.com/highcharts.js

#### Affected browser(s)
All

**Comment by pawelfus** [comment]
Thanks for reporting the issue @soetji 

In general, you shouldn't update chart in `chart.render` event which creates infinite loops (it simply fires `render` over and over again). Boost is designed to improve performance, while updating yAxis (e.g. it's title) requires extra chart re-render so in result chart rendering might be slower.

You can try one of the solutions below, which base on internal methods and props - this is fast and does't require re-rendering whole chart. Downside: if your yAxis text can be quite long and you need e.g. two lines etc it might be problematic (because title should re-render the whole chart again..).

-  https://jsfiddle.net/BlackLabel/2umko6Le/1/ 
```js
function yAxesTitle() {
  this.yAxis.forEach(yAx => {
    const text = 'test';
    if (yAx.options.title.text !== text) {
      yAx.options.title.text = text;
      yAx.axisTitle = yAx.axisTitle.destroy();
      yAx.addTitle(true);
      yAx.axisTitle.attr(yAx.getTitlePosition());
    }
  });
}
```

- https://jsfiddle.net/BlackLabel/2umko6Le/ 
```js
function yAxesTitle() {
  this.yAxis.forEach(yAx => {
    const text = 'test';
    if (yAx.options.title.text !== text) {
      yAx.axisTitle.attr({ text });
    }
  });
}
```

**Comment by soetji** [comment]
Thanks @pawelfus for your comment. I realize about the possible infinite loop in `chart.render`, hence the conditional call to the `yAx.setTitle()`. Still though, per your advice, I would like to avoid this if I can. So is there a better way to update the axis title based on series update (adding or removing)? In my case I am displaying the data unit on the yAxis and depending on the series being displayed the unit can change.

Thanks also on your solution above. It works on jsfiddle. However on my real code I getting an error of `yAx.axisTitle` undefined. Please advice.

**Comment by pawelfus** [comment]
> So is there a better way to update the axis title based on series update (adding or removing)?

When you add/remove series, call `yAxis.setTitle` instead of using `render:
```js
chart.addSeries(options, false); // disable redraw
chart.yAxis[0].setTitle({ text: newText }); // enable redraw
```

> Thanks also on your solution above. It works on jsfiddle. However on my real code I getting an error of yAx.axisTitle undefined. Please advice.

Perhaps on init title is disabled? Any text on init should be fine.

**Comment by soetji** [comment]
@pawelfus Thanks for your comment. I forgot to mention that I have to base the yAxis title on the current max tick value of the axis. It seems that that value is available only after render or redraw. So `chart.addSeries(options, false)` is not useful for I need. Any other thoughts would be very much appreciated.

I got the `yAx.axisTitle.attr({ text })` working by adding a initial title text of `' '` (i.e, a space). Previously I had an empty string and that did not initialize the title.

**Comment by pawelfus** [comment]
Right, good point. Then perhaps an option like `title.formatter` would be a nice solution for you, take a look: https://jsfiddle.net/BlackLabel/Lm510gce/

**Comment by soetji** [comment]
@pawelfus That formatter works perfectly for my project. It works in the boost and non-boost mode. It doesn't cause extra redrawing. The final rendering looks correct. It is safer than what I did with calling `axis.setTitle()` in the `chart.render` callback. Great job! Thank you!

Will it make it to the official releases?

**Comment by pawelfus** [comment]
I think it's a good idea, but requires more tests. We need to make sure that reserved space for title is correct, does work with redraws/resizes, multiple lines etc. Perhaps we can consider this in v8, not sooner. It would be good to gather more feedback - could you create a ticket in our [UserVoice](https://highcharts.uservoice.com/forums/55896-highcharts-javascript-api)?

FYI @TorsteinHonsi

**Comment by soetji** [comment]
It's been a while. Has this issue been fixed and released?

**Comment by KacperMadej** [comment]
The bug is still visible - it is not fixed and the issue will be reopened.
A workaround is available and seems to be working nicely.
In case you have opened a UserVoice ticket for the mentioned above enhancement please add the link in a comment.

**Comment by raf18seb** [comment]
Simplified demo: https://jsfiddle.net/BlackLabel/bamdopyh/

The issue is visible when trying to redraw the boosted chart with at least 2 series.

**Comment by ochristensen** [comment]
in our scenario the workaround above is not possible; we a) move axis labels based on rendered dimensions of several parts of the graph and b) specify plotlines at axis extremes (which we *might* be able to do in the config but is less ideal for our purposes) .  is there an alternate workaround?

**Comment by raf18seb** [comment]
@ochristensen unfortunately, we don't have any workaround for the `chart.redraw()` issue.

But we can suggest moving your labels similarly like pawelfus' solution (by changing labels' `attr`).
The same with plotLines: you can  move them by setting `attr` instead of updating or you can render them using the Highcharts SVG Renderer tool without any plotLines config and chart redrawing.

Please contact our support engineers in case you need a help with it or you have any further questions: https://www.highcharts.com/blog/support/

**Comment by soetji** [comment]
Hello, with all the updates you have made since I opened this ticket, any plan on fixing this now? :)

**Comment by soetji** [comment]
Has the workaround of adding an option of `title.formatter` made it into release?

**Comment by karolkolodziej** [comment]
No, it did not make it into an official release. 
Could you please @soetji [create a separate feature request](https://github.com/highcharts/highcharts/issues/new/choose) for it? Because as you can see it got lost in the noise here.
